### PR TITLE
adding iir component to xperi topology

### PR DESCRIPTION
**Pull request by rosschisholm-xperi**
Signed-off-by: Ross Chisholm <ross.chisholm@xperi.com>

### CODE REVIEW

File: tools/topology/CMakeLists.txt [hunk 1]
```diff
@@ -122,6 +122,7 @@ set(TPLGS
 	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le"
 	"sof-tgl-max98357a-rt5682\;sof-tgl-rt1011-rt5682\;-DCODEC=RT1011\;-DFMT=s24le"
 	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682\;-DAMP_SSP=1"
+	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682-xperi\;-DAMP_SSP=1\;-DXPERI=1"
```
Comment (lgirdwood) [inline tools/topology/CMakeLists.txt:125]: Please have a look at the format and of other commit messages to align the subject. The commit message should also say why you are making this change e.g.
```
topology: add support for xperi processing feature X
This patch adds IIR support as it is required to as a dependecy to enable xperi feature X....

Signed-off-by:....
```

File: tools/topology/sof-tgl-max98373-rt5682.m4 [hunk 2]
```diff
@@ -94,10 +94,21 @@ dnl     frames, deadline, priority, core)
...
-PIPELINE_PCM_ADD(sof/pipe-volume-playback.m4,
+ifelse(XPERI, `1',
+`
+PIPELINE_PCM_ADD(sof/pipe-eq-iir-volume-playback.m4,
         2, 1, 2, s32le,
 	1000, 0, 0,
 	48000, 48000, 48000)
+'
+,
+`
```
Comment (lgirdwood) [inline tools/topology/sof-tgl-max98373-rt5682.m4:105]: Needs a comment for each item you are adding here.

**Review by lgirdwood — CHANGES_REQUESTED** [review]
Needs a little more detail in commit message and comments.

File: tools/topology/CMakeLists.txt (hunk 1 above)
Comment (rosschisholm-xperi) [reply tools/topology/CMakeLists.txt:125]: Updated commit message as described

File: tools/topology/sof-tgl-max98373-rt5682.m4 (hunk 2 above)
Comment (rosschisholm-xperi) [reply tools/topology/sof-tgl-max98373-rt5682.m4:105]: Added comment for pipeline that was missing

File: tools/topology/CMakeLists.txt [hunk 3]
```diff
@@ -122,6 +122,7 @@ set(TPLGS
 	"sof-tgl-max98357a-rt5682\;sof-tgl-max98357a-rt5682\;-DCODEC=MAX98357A\;-DFMT=s16le"
```
Comment (dbaluta) [inline tools/topology/CMakeLists.txt:122]: @rosschisholm-xperi would be nice to add more details in the commit message for the reader to better understand the context. For example what are Xperi tunings more exactly.

File: tools/topology/CMakeLists.txt (hunk 1 above)
Comment (cujomalainey) [reply tools/topology/CMakeLists.txt:125]: given we ware using generic components mind changing the flag name to something generic as well like "INCLUDE_EQ"?

Comment (rosschisholm-xperi) [reply tools/topology/CMakeLists.txt:125]: Yes I think this is a good idea - I'll update it. 
Generally what is the strategy for topologies in the long term. Surely as more devices are made that all target the same topology if there's different requirements in terms of the components/pipelines then the topology is going to become more complicated

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
Please use "main" as PR target.

**Comment by cujomalainey** [comment]
@rosschisholm-xperi please resend pr with `main` as new target and link here

**Comment by rosschisholm-xperi** [comment]
@cujomalainey - submitted yesterday, here's the link: https://github.com/thesofproject/sof/pull/3967

**Pull request closed without merging**